contracts/include/blackjack/blackjack.hpp

```diff
@@ -214,8 +215,15 @@ class [[eosio::contract]] blackjack: public game_sdk::game {
         }
         // remove player's cards
         clean_labels(multideck, state_itr);
-        service::shuffle(multideck.begin(), multideck.end(), get_prng(std::move(rand)));
-        return multideck;
+        // draw 9 cards
+        card_game::labels_t result(9);
+        auto prng = get_prng(std::move(rand));
+        for (int i = 0; i < 9; i++) {
+            const auto idx = prng->next() % multideck.size();
+            result[i] = multideck[idx];
+            multideck.erase(multideck.begin() + idx);
```

Comment:
Looks like `labels.size()` = (13\*4)\*8 = 416. `std::vector::erase()` performs O(size/2) copy/move operations in average (due to uniform distribution). For 9 iterations it will be 416/2\*9 = 1872 ops.

Think about the following variant:
```
size_t last;
for (size_t i = 0; i < 9; i++) {
    last = multideck.size() - i;
    const auto idx = prng->next() % last;
    std::swap(multideck[idx], multideck[last - 1]);
}￼
card_game::labels_t result{multideck.begin() + last - 1, multideck.end()};
```